test/e2e/conftest.py

```diff
@@ -15,13 +16,27 @@
 
 E2E_TEST_DIR = Path(__file__).parent
 REPOSITORY_ROOT_DIR = E2E_TEST_DIR.parent.parent
-BIGQUERY_PROFILES_TEMPLATE_FILE = E2E_TEST_DIR / "profiles_template_bigquery.yaml"
+
 BIGQUERY_PROJECT_NAME = "layer-bigquery"
-BIG_QUERY_CREDENTIALS = os.getenv("BIG_QUERY_CREDENTIALS") or ""
+BIGQUERY_CREDENTIALS = os.getenv("BIGQUERY_CREDENTIALS") or "{}"
+
+SNOWFLAKE_PROJECT_NAME = "layer-snowflake"
+SNOWFLAKE_CREDENTIALS = os.getenv("SNOWFLAKE_CREDENTIALS") or "{}"
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    parser.addoption("--adapter", action="store", default="bigquery")
```

Comment:
Do we want a default? Might be confusing for people used to calling `make e2e-test` not knowing that it is now parametrized.

I would also expect if I run `make e2e-test` to run all the tests, not half of them. Is there a specific reason you decided to have a switch rather than always run them all?

Reply:
It was because the extras need to be installed that I didn't want to make it default to running them all. That being said, I think it's a good idea to make this a required param and fail if it's not passed in.

Reply:
Made this such that by default it tests all adapters, but can be filtered with the `--adapter` argument.